agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4424,8 +4424,8 @@ bool slave_thread::handle_autoconfiguration_wsc(Socket *sd, ieee1905_1::CmduMess
     // way to distinguish them: the M1 message has the AP_Radio_Basic_Capabilities TLV,
     // while the M2 has the AP_Radio_Identifier TLV.
     // If this is a looped back M1 CMDU, we can treat is as handled successfully.
-    if (cmdu_rx.getNextTlvType() == int(wfa_map::eTlvTypeMap::TLV_AP_RADIO_BASIC_CAPABILITIES)) {
-        LOG(DEBUG) << "looped back M1 CMDU";
+    if (cmdu_rx.hasTlv(uint8_t(wfa_map::eTlvTypeMap::TLV_AP_RADIO_BASIC_CAPABILITIES))) {
+        LOG(DEBUG) << "agent M1 CMDU, ignore";
```

Comment:
I would feel more comfortable if we did the same in the agent as in the controller. It's not that complicated, right?

I realize that it's a bit of wasted effort, since with the full TLV list parsing that @Galvien is working on, it will no longer be needed. But we only need to handle for types of TLVs so it's just about 20 lines of code...

Note that with that, there is no need for hasTlv at all any more. Which means that all my comments on it no longer apply :-)

Reply:
You are absolutely right.

Reply:
If hasTlv isn't needed in this PR in the end, just push it to a new branch without PR to save it for later, and remove it from this PR.

Reply:
OK - done, see #232